pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -238,7 +248,19 @@ func (rc *reconciler) mountAttachVolumes() {
 				}
 			}
 		} else if !volMounted || cache.IsRemountRequiredError(err) {
-			// Volume is not mounted, or is already mounted, but requires remounting
+			if !volMounted && rc.podStateProvider.ShouldPodContainersBeTerminating(volumeToMount.Pod.UID) {
+				// the volume is not yet mounted, and the pod containers should be terminating, so don't try to set up a new mount
+				// ShouldPodContainersBeTerminating: Intended for use by subsystem sync loops to avoid performing background setup after termination has been requested for a pod.
+				klog.V(4).InfoS(volumeToMount.GenerateMsgDetailed("Skipping operationExecutor.MountVolume because pod containers should be terminating", ""), "pod", klog.KObj(volumeToMount.Pod))
+				continue
+			}
+			if rc.podStateProvider.ShouldPodRuntimeBeRemoved(volumeToMount.Pod.UID) {
+				// no containers are running and we should be cleaning up, so don't try to set up a new mount or remount
+				// "ShouldPodRuntimeBeRemoved ... is true when a pod is not yet observed by a worker after the first sync (meaning it can't be running yet) or after all running containers are stopped."
+				klog.V(4).InfoS(volumeToMount.GenerateMsgDetailed("Skipping operationExecutor.MountVolume remount because pod runtime should be removed", ""), "pod", klog.KObj(volumeToMount.Pod))
+				continue
+			}
```

Comment:
I'm more confident in this block. This should resolve the "not registered" error seen because of a mount attempt after `syncTerminatedPod` was called